base.py

```diff
@@ -69,3 +77,39 @@ def _validate_input(self, X: List):
 
         X = [MolFromSmiles(x) if type(x) == str else x for x in X]
         return X
+
+
+class FingerprintGeneratorMixin(ABC):
+    def __init__(self, fingerprint_type: str = "bit", sparse=False):
+        """
+        result_vector_tape has to be one of the following:
+        bit, spares, count, sparse_count
+        """
+        assert fingerprint_type in ["bit", "count"]
+        self.result_vector_type = fingerprint_type
+        self.fp_generator_kwargs = {}
+        self.sparse_output = sparse
+
+    @abstractmethod
+    def _get_generator(self):
+        """
+        Function that creates a generator object in each sub-process.
+
+        :return: rdkit fingerprint generator for current fp_generator_kwargs parameter
+        """
+        pass
+
+    def _generate_fingerprints(
+        self, X: Union[pd.DataFrame, np.ndarray]
+    ) -> Union[np.ndarray, sparse.csr_array]:
+        fp_generator = self._get_generator()
+
+        if self.result_vector_type == "bit":
+            X = [fp_generator.GetFingerprintAsNumPy(x) for x in X]
+        else:
+            X = [fp_generator.GetCountFingerprintAsNumPy(x) for x in X]
+
+        if self.sparse_output:
+            return sparse.csr_array(X)
+        else:
+            return np.array(X)
```

Comment:
I don't get why we can't just have this method in `FingerprintTransformer`. We have everything we need there, I think. I mean, we have information about bit/count, about dense/sparse, and `_get_generator()` was there before anyway.

I get that not all fingerprints have the generator interface, but those ones can just override the `_generate_fingerprints()`, right?